src/main/java/dev/streamx/sling/connector/impl/DefaultHttpClientFactory.java

```
@@ -1,46 +1,115 @@
+    this.config.set(config);
+  }
+
+  private Optional<LayeredConnectionSocketFactory> withTrustAllTlsCertificates() {
+    try {
+      LOG.trace("Building {}", LayeredConnectionSocketFactory.class.getName());
+      TrustStrategy trustStrategy = (chain, authType) -> true;
+      SSLContext sslContext = SSLContextBuilder.create()
+          .loadTrustMaterial(null, trustStrategy)
+          .build();
+      return Optional.of(new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE));
+    } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException exception) {
+      String message = String.format(
+          "Could not create %s", LayeredConnectionSocketFactory.class.getName()
+      );
+      LOG.error(message, exception);
+      return Optional.empty();
+    }
   }
 
-  public CloseableHttpClient createNewClient() {
-    PoolingHttpClientConnectionManager connMgr = new PoolingHttpClientConnectionManager();
-    connMgr.setMaxTotal(config.max_total());
-    connMgr.setDefaultMaxPerRoute(config.max_per_route());
+  private PoolingHttpClientConnectionManager connectionManager() {
+    boolean doTrustAllTlsCerts = config.get().do$_$trust$_$all$_$tls$_$certs();
+    PoolingHttpClientConnectionManager connMgr;
+    if (doTrustAllTlsCerts) {
+      connMgr = withTrustAllTlsCertificates()
+          .map(sslSocketFactory -> {
+                PlainConnectionSocketFactory plainSocketFactory
+                    = PlainConnectionSocketFactory.getSocketFactory();
+                return RegistryBuilder.<ConnectionSocketFactory>create()
+                    .register("http", plainSocketFactory)
+                    .register("https", sslSocketFactory)
+                    .build();
+              }
+          ).map(registry -> {
+            LOG.debug("Using custom connection manager");
+            return new PoolingHttpClientConnectionManager(registry);
+          })
+          .orElse(new PoolingHttpClientConnectionManager());
```

Looks like in some cases even though `doTrustAllTlsCerts` is true - a standard connection manager can be returned (in this line) - the same one as in the `else` branch.

Maybe we should throw an exception if we are unable to create a connection manager as the caller asked us for?

---

Good point. Changed!